src/components/Patient/PatientRegistration.tsx

```diff
@@ -162,142 +193,92 @@ export default function PatientRegistration(
         query: {
-          phone_number: form.phone_number,
+          phone_number: form.getValues("phone_number"),
         },
       });
     } else {
       setSuppressDuplicateWarning(true);
     }
   };
 
-  const handleFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    const validate = validatePatient(form, ageDob === "dob");
-    if (typeof validate !== "object") {
-      patientId
-        ? updatePatientMutation.mutate({ ...mutationData, ward_old: undefined })
-        : createPatientMutation.mutate({
-            ...mutationData,
-            facility: facilityId,
-            ward_old: undefined,
-          });
-    } else {
-      const firstErrorField = document.querySelector("[data-input-error]");
-      if (firstErrorField) {
-        firstErrorField.scrollIntoView({ behavior: "smooth", block: "center" });
-      }
-      Notification.Error({
-        msg: t("please_fix_errors"),
-      });
-      setFeErrors(validate);
+  const patientPhoneSearch = useQuery({
+    queryKey: ["patients", "phone-number", debouncedNumber],
+    queryFn: query(routes.searchPatient, {
+      body: {
+        phone_number: parsePhoneNumber(debouncedNumber || "") || "",
+      },
+    }),
+    enabled: !!parsePhoneNumber(debouncedNumber || ""),
+  });
+
+  const duplicatePatients = useMemo(() => {
+    return patientPhoneSearch.data?.results.filter((p) => p.id !== patientId);
+  }, [patientPhoneSearch.data, patientId]);
+
+  const patientQuery = useQuery({
+    queryKey: ["patient", patientId],
+    queryFn: query(routes.getPatient, {
+      pathParams: { id: patientId || "" },
+    }),
+    enabled: !!patientId,
+  });
+
+  useEffect(() => {
+    if (patientQuery.data) {
+      form.reset({
+        ...patientQuery.data,
+        same_phone_number:
+          patientQuery.data.phone_number ===
+          patientQuery.data.emergency_phone_number,
+        same_address:
+          patientQuery.data.address === patientQuery.data.permanent_address,
+        yob_or_dob: patientQuery.data.date_of_birth ? "dob" : "age",
+        geo_organization: (
+          patientQuery.data.geo_organization as unknown as Organization
+        )?.id,
+      } as unknown as z.infer<typeof formSchema>);
     }
-  };
+  }, [patientQuery.data]); // eslint-disable-line react-hooks/exhaustive-deps
 
   useEffect(() => {
     const handler = setTimeout(() => {
-      if (!patientId || patientQuery.data?.phone_number !== form.phone_number) {
+      const phoneNumber = form.getValues("phone_number");
+      if (!patientId || patientQuery.data?.phone_number !== phoneNumber) {
         setSuppressDuplicateWarning(false);
       }
-      setDebouncedNumber(form.phone_number);
+      setDebouncedNumber(phoneNumber);
     }, 500);
 
     return () => {
       clearTimeout(handler);
     };
-  }, [form.phone_number]);
+  }, [form.watch("phone_number")]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
assign this to a variable and reference that here? otherwise it'd keep re-rendering right? (not sure)

Reply:
This shouldn't be a problem, because watch is used to trigger rerender when ever there are change